server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -99,10 +100,14 @@ public MailboxBuilder builder() {
     private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<MailboxMetaData>> userMailboxesMetadata,
                                                  MailboxSession mailboxSession) throws MailboxException {
         MailboxPath mailboxPath = messageManager.getMailboxPath();
+        boolean isOwner = isSameUser(mailboxSession, mailboxPath);
         Optional<Role> role = Role.from(mailboxPath.getName());
         MailboxCounters mailboxCounters = messageManager.getMailboxCounters(mailboxSession);
         MessageManager.MetaData metaData = messageManager.getMetaData(NO_RESET_RECENT, mailboxSession, MessageManager.MetaData.FetchGroup.NO_COUNT);
 
+        Rights rights = Rights.fromACL(metaData.getACL());
+        Username username = new Username(mailboxSession.getUser().getUserName());
```

Comment:
jmap Username should be able to build itself from a User, don't you think ?

Reply:
I think you can even say "from a mailboxSession"